url71665d7d68_branscha_lib_scripty_tgz-pJ8-com_sdicons_scripty_ScriptyAutoReplJ8/src/com/sdicons/scripty/cmdlib/BeanLibrary.java

```diff
@@ -0,0 +1,560 @@
+        return lResult;
+    }
+
+    @SuppressWarnings("unchecked")
+    private void update(Object aTarget, Object aProp, Object aVal)
+            throws CommandException
+    {
+        String lPropName = "";
+        int lIdx = -1;
+
+        if(aProp instanceof String) lPropName = (String) aProp;
+        else throw new CommandException("Don't know what to do with non-string keys ... (for now).");
+
+        try
+        {
+            // Try to interprete the property as
+            // an integer.
+            lIdx = Integer.parseInt(lPropName);
+        }
+        catch (NumberFormatException e1)
+        {
+            // Ignore this exception!
+        }
+
+        if(aTarget instanceof List)
+        {
+            final List lTargetList = (List) aTarget;
+            if(lIdx < 0)
+                throw new CommandException("Modifying list should use an index.");
+            if(lIdx >= lTargetList.size())
+                throw new CommandException("Array index out of range.");
+
+            // TODO: catch errors on this one.
+            lTargetList.set(lIdx, aVal);
+
+        }
+        else if(aTarget != null)
+        {
+            if(aTarget.getClass().isArray())
+            {
+                // Array access.
+                if(lIdx < 0)
+                    throw new CommandException("Modifying array should use an index.");
+                if(lIdx >= Array.getLength(aTarget))
+                    throw new CommandException("Array index out of range.");
+
+                // TODO: catch errors on this one.
+                Array.set(aTarget, lIdx, aVal);
+            }
+            else
+            {
+                try
+                {
+                    // JavaBean access.
+                    BeanInfo lInfo = Introspector.getBeanInfo(aTarget.getClass());
+                    PropertyDescriptor lProps[] = lInfo.getPropertyDescriptors();
+                    for (PropertyDescriptor lProp : lProps)
+                    {
+                        if(lProp.getDisplayName().equals(lPropName))
+                        {
+                            Method lMeth = lProp.getWriteMethod();
+                            if(lMeth == null)
+                                throw new CommandException("Property cannot be written.");
+
+                            if(lProp.getPropertyType() != aVal.getClass() && aVal instanceof String)
+                            {
+                                PropertyEditor lEditor = lProp.createPropertyEditor(aTarget);
+                                lEditor.setAsText((String) aVal);
+                            }
+                            else lMeth.invoke(aTarget, new Object[]{aVal});
+                        }
+                    }
+                }
+                catch (Exception e)
+                {
+                    throw new CommandException(e.getMessage(), e);
+                }
+            }
+        }
+    }
+
+    private void render(Object aObj, PrintWriter aWriter)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 15. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.